### PR TITLE
[DOC] Format API docs for Typescript - @ember/array

### PR DESCRIPTION
**Pull request by shibulijack**
**Issue**: https://github.com/emberjs/ember.js/issues/20172.

**Details**:
- Moved comments from Mixin.create({...}) to interface.
- Added missing interface property for `[]` in `array/index`.
- Added interface properties for `array/proxy`

### CODE REVIEW

**Review by ttbach — APPROVED** [review]
lgtm

File: packages/@ember/array/index.ts [hunk 1]
```diff
@@ -250,86 +250,13 @@ function mapBy<T>(this: EmberArray<T>, key: string) {
...
-  },
-
   /**
     __Required.__ You must implement this method to apply this mixin.
 
     Your array must support the `length` property. Your replace methods should
     set this property whenever it changes.
-
-    @property {Number} length
-    @public
```
Comment (chriskrycho) [inline packages/@ember/array/index.ts:330]: We need to preserve these annotations in the switchover! They're used by the doc generation tool in knowing (a) what to emit (`@public`) and (b) how to render it in the docs (the type, in particular).

File: packages/@ember/array/index.ts [hunk 2]
```diff
@@ -1552,18 +1328,8 @@ const MutableArray = Mixin.create(EmberArray, MutableEnumerable, {
     colors.insertAt(2, 'yellow');  // ['red', 'green', 'yellow', 'blue']
     colors.insertAt(5, 'orange');  // Error: Index out of range
     ```
-
-    @method insertAt
-    @param {Number} idx index of insert the object at.
-    @param {Object} object object to insert
-    @return {EmberArray} receiver
-    @public
```
Comment (chriskrycho) [inline packages/@ember/array/index.ts:1560]: Same for the annotations for this mixin!

File: packages/@ember/array/proxy.ts [hunk 3]
```diff
@@ -169,30 +196,10 @@ class ArrayProxy<T, C extends EmberArray<T> | T[] = T[]>
     this._removeArrangedContentArrayObserver();
   }
 
-  /**
-    The content array. Must be an object that implements `Array` and/or
-    `MutableArray.`
-
-    @property content
-    @type EmberArray
-    @public
```
Comment (chriskrycho) [inline packages/@ember/array/proxy.ts:178]: Same for these.

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Overall looks good; there's a miss here on needing to copy over the various doc annotations as well, but otherwise 👍🏼 – thanks. Let me know when you've updated for the annotations, and I'll re-review and we'll merge!

File: packages/@ember/array/index.ts (hunk 1 above)
Comment (shibulijack) [reply packages/@ember/array/index.ts:330]: Thanks for clarifying, Chris. This PR initially had those annotations earlier but I removed it later after referring to similar PRs. I'll add them back now.

Comment (chriskrycho) [reply packages/@ember/array/index.ts]: Yeah, not your fault at all—thank you!

**Review by chriskrycho — APPROVED** [review]
Thanks for making the changes!

File: packages/@ember/array/index.ts [hunk 4]
```diff
@@ -1224,29 +1021,17 @@ const EmberArray = Mixin.create(Enumerable, {
       }); // false (true && false && false)
     ```
 
-    @method reduce
-    @param {Function} callback The callback to execute
-    @param {Object} initialValue Initial value for the reduce
-    @return {Object} The reduced value.
+    @method isAny
```
Comment (chriskrycho) [inline packages/@ember/array/index.ts:1024]: Whoops, ended up with the wrong doc comment annotations copied over here—you have the `isAny` annotations on `reduce`!

**Review by chriskrycho — CHANGES_REQUESTED** [review]
One miss!

File: packages/@ember/array/index.ts (hunk 4 above)
Comment (shibulijack) [reply packages/@ember/array/index.ts]: Oops! 😅 
Fixed it.

**Review by chriskrycho — APPROVED** [review]

**Pull request merged**